ggshield/pypi.py

```diff
@@ -0,0 +1,109 @@
+import re
+import shutil
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Dict, List, Set
+
+import click
+
+from ggshield.config import Config
+from ggshield.output import OutputHandler
+from ggshield.path import get_files_from_paths
+from ggshield.scan import Files, Result, ScanCollection
+from ggshield.utils import SupportedScanMode
+
+
+PYPI_DOWNLOAD_TIMEOUT = 30
+
+
+def save_package_to_tmp(temp_dir: str, package_name: str) -> None:
+    command: List[str] = [
+        "pip",
+        "download",
+        package_name,
+        "--dest",
+        temp_dir,
+        "--no-deps",
+    ]
+
+    try:
+        click.echo("Downloading pip package... ", nl=False)
+        subprocess.run(
+            command,
+            check=True,
+            stderr=subprocess.PIPE,
+            timeout=PYPI_DOWNLOAD_TIMEOUT,
+        )
+        click.echo("OK")
+
+    except subprocess.CalledProcessError:
+        raise click.ClickException(f'Package "{package_name}" not found')
+
+    except subprocess.TimeoutExpired:
+        raise click.ClickException('Command "{}" timed out'.format(" ".join(command)))
+
+
+def get_files_from_package(
+    temp_dir: str,
+    package_name: str,
+    exclusion_regexes: Set[re.Pattern],
+    verbose: bool,
+) -> Files:
+    temp_dir_path: Path = Path(temp_dir)
+    archive: Path = list(temp_dir_path.glob("*"))[0]
```

Comment:
could be `next(temp_dir_path.iterdir())`, but I think it would be nicer to have `save_package_to_tmp()` do this and return the path.

Reply:
Done for the use of `next...`. 

I feel the current code split into the 2 functions is clearer and easier to test like that, `save_package_to_tmp` doing only the command call in subprocess and `get_files_from_package` handling the "logic".